pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -172,14 +174,365 @@ public static String javaPrimitiveToObjectType(String primitiveFieldType) {
 		};
 	}
 
+	/**
+	 * Recursively calculates the hashcode for a message fields.
+	 *
+	 * @param fields The fields of this object.
+	 * @param generatedCodeSoFar The accumulated hash code so far.
+	 * @return The generated code for getting the hashCode value.
+	 */
+	public static String getFieldsHashCode(final List<Field> fields, String generatedCodeSoFar) {
+		for(Field f : fields) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsHashCode(oneOfField.fields(), generatedCodeSoFar);
+			}
```

Comment:
I don't know much about OneOf fields, so the question is: can a OneOf field be optional/repeated? I mean, should line 191 below be just "if" or "else if"?

Reply:
Yes, you're right. It has to be mutually exclusive.